pkgs/build-support/testers/default.nix

```
@@ -57,12 +57,15 @@
   # or doc/builders/testers.chapter.md
   testVersion =
     { package,
-      command ? "${package.meta.mainProgram or package.pname or package.name} --version",
+      parameter ? "--version",
+      executable ? package.meta.mainProgram or package.pname or package.name,
+      command ? "${executable} ${parameter}",
       version ? package.version,
+      allowFailure ? false,
```

If it's an unexpected kind of failure, it's better to stop immediately. For instance, the program might crash, but also print its version in the crash report. By checking for a specific status, we have a chance to catch that scenario.

```suggestion
      exitStatus ? 0,
```